osbs/utils/yaml.py

```diff
@@ -50,12 +53,22 @@ def read_yaml(yaml_data, schema, package=None):
         logger.error('unable to extract JSON schema, cannot validate')
         raise
 
+    # Load schema into py object
     try:
         schema = json.load(schema)
     except ValueError:
         logger.error('unable to decode JSON schema, cannot validate')
         raise
-    data = yaml.safe_load(yaml_data)
+
+    validate_with_schema(data, schema)
+    return data
+
+
+def validate_with_schema(data, schema):
+    """
+    :param data: object, data to be validated
```

Comment:
Type of `data` could be more specific here.

Reply:
It's whatever `yaml.safe_load()` returns; IIUC, it's a Python Dict. I'll modify the comment.